crates/superposition_types/src/overridden.rs

```diff
@@ -0,0 +1,33 @@
+use std::collections::HashSet;
+
+use serde_json::{Map, Value};
+
+use crate::config::Overrides;
+
+pub(crate) fn filter_keys_by_prefix(
+    overrides: Map<String, Value>,
+    prefix_list: &HashSet<String>,
+) -> Map<String, Value> {
+    overrides
+        .into_iter()
+        .filter(|(key, _)| {
+            prefix_list
+                .iter()
+                .any(|prefix_str| key.starts_with(prefix_str))
+        })
+        .collect()
+}
```

Comment:
Should this be an implementation on struct Config as well, since it does have configs. We can probably do 
```
get_resolved_config(....).filter_config_keys_by_prefix(...)
```

Reply:
for `Config` type, it requires a custom behaviour,
you would always want to perform the filtering on the individual contexts first, so that the amount of data being processed on by `jsonlogic` can be reduced to provide faster execution, as compared to resolving on the entire data and then filtering, as the eval operation in `jsonlogic` is extremely slow